com.unity.netcode.gameobjects/Utilities/TestHelpers/Runtime/NetcodeIntegrationTest.cs

```diff
@@ -0,0 +1,681 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+using Object = UnityEngine.Object;
+
+namespace Unity.Netcode.TestHelpers.Runtime
+{
+    /// <summary>
+    /// The default Netcode For GameObjects integration test helper class
+    /// </summary>
+    public abstract class NetcodeIntegrationTest
```

Comment:
one problem I have with this PR is that it does multiple things which is too much.
for instance, I'm looking at MTT-2435's title & description: "Can not test individual test assemblies without RuntimeTests assembly"
this PR is doing much more than that, it does rename & refactor too, even some CI changes and so on.
I think we need to split this PR into more focused PR chunks and get them in separately rather than in a one single big lump sum.

Reply:
We definitely should discuss this as splitting this up will cause issues.
I was actually going to rename MTT-2435 to be better suited to the changes.
This PR also ended up fixing the issues with [PR-1665 ](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/pull/1665) as well as making sure that all tests back ported from experimental to develop passed on consoles.

We can split it up, but it could become complicated at this point.

Reply:
can't we start with new asmdef without any fixes, renames or any other changes first?
later, we could work on renames then fixes specifically, no?

Reply:
We could... but the longer we wait the more things will need to be updated and the more integration tests will be written the legacy way (which means more things to update).  
But we can definitely break this all out over a span of time for sure.